src/M2ulPhyS.cpp

```diff
@@ -258,20 +298,28 @@ void M2ulPhyS::initVariables() {
 
   max_char_speed = 0.;
 
-  // NOTE: num_equations to be given by the gasMixture
-  switch (eqSystem) {
-    case EULER:
-      num_equation = 2 + nvel;
-      break;
-    case NS:
-      num_equation = 2 + nvel;
-      break;
-    case NS_PASSIVE:
-      num_equation = 2 + nvel + 1;
-      break;
-    default:
-      break;
-  }
+  // Kevin: if mixture is set to a preset such as DRY_AIR,
+  // it will not read the input options for following variables,
+  // instead applies preset values.
+  numSpecies = mixture->GetNumSpecies();
+  numActiveSpecies = mixture->GetNumActiveSpecies();
+  ambipolar = mixture->IsAmbipolar();
+  twoTemperature_ = mixture->IsTwoTemperature();
+  num_equation = mixture->GetNumEquations();
```

Comment:
@dreamer2368 : Will this handle the axisymmetric case (i.e., when the number of velocities is 3 but the number of spatial dimensions is 2) properly?

Reply:
For `DryAir`, it works by simply initializing `mixture` with `nvel` instead of `dim`.
For `PerfectMixture`, at least two routines require some minor fixes:
- `computeElectronPressureGrad`
- `computeConservedStateFromConvectiveFlux`
`GasMixture` will eventually need to take both `dim` and `nvel` to avoid confusion in future.
These, including other necessary changes for axisymmetric solver, will be resolved at the branch `axis_mixture`.